Energy line test (#669)
avaframe/AvaFrame: docs/moduleAna1Tests.rst

            :width: 90%

            Pile after 19.8s

Energy line test
==================

> The Energy line test compares the results of the DFA simulation to the geometrical solution that is related
> to the total energy of the system. Solely considering coulomb friction this solution is motivated by the first principle
> of energy conservation along a simplified topography. Here friction force only depends on the slope angle.
> The analytical run-out is the intersection of the path profile with the so-called alpha line defined by the friction angle.
> It is also possible to extract from this alpha line information about the flow mass averaged velocity at any time or
> position along the path profile.

Reviewer: ```suggestion
The Energy line test compares the results of the DFA simulation to a geometrical solution that is related
to the total energy of the system. Solely considering coulomb friction this solution is motivated by the first principle
of energy conservation along a simplified topography. Here friction force only depends on the slope angle.
The analytical run-out is the intersection of the path profile with the alpha line defined by the friction angle.
From the alpha line it is also possible to extract information about the flow mass averaged velocity at any time or
position along the path profile.
```
Please add `slope angle` `friction angle` and `alpha line` to the glossary

Theory
-------

> Applying the energy conservation law to a material bloc flowing down
> a slope and obeying to Coulomb friction and this between two
> infinitesimally close time steps reads:

Reviewer: ```suggestion
Applying the energy conservation law to a material block flowing down
a slope with Coulomb friction and this between two
infinitesimally close time steps reads:
```

.. math::
    \begin{aligned}
    E^{tot}_{t + dt} - E^{tot}_{t} & = E^{kin}_{t + dt} + E^{pot}_{t + dt}
    - (E^{kin}_{t} + E^{pot}_{t}) = \delta E_{fric} \\
    & = \frac{1}{2} m v^2_{t+dt} + m g z_{t+dt}
    - \frac{1}{2} m v^2_{t} - m g z_{t} \\
    &= \mathbf{F}_{fric} . \mathbf{dl} =
    - \mu \vert \vert \mathbf{N}\vert \vert \frac{\mathbf{dl}}{dl}. \mathbf{dl}
    = - \mu mg (\mathbf{e_z}.\mathbf{n}) dl
    \end{aligned}

where :math:`\delta E_{fric}` is the energy dissipation due to friction,
:math:`\mathbf{N}` represents the normal (to the slope surface) component
of the gravity force, :math:`\mathbf{n}` the normal vector to the slope surface
and :math:`\mathbf{dl}` is the vector representing the distanced traveled by
the material between :math:`t` and :math:`t+dt`.
The normal vector reads :math:`\mathbf{e_z}.\mathbf{n} = cos(\theta)`,
where :math:`\theta` is the slope angle.
Finally, in the 2D case, :math:`dl = \frac{ds}{cos(\theta)}`, which means that
the material is flowing in the steepest slope direction (:math:`\mathbf{ds}`
is the horizontal component of :math:`\mathbf{dl}`).

> 

Reviewer: I think the non described variable (e.g. m, mu, g, etc) should be at least named in parentheses at the end of this paragraph

Integrating the energy conservation between the start and a time t reads:

.. math::
    \begin{aligned}
    E^{tot}_{t} - E^{tot}_{t=0} & = \frac{1}{2} m v^2_{t} + m g z_{t}
    - \frac{1}{2} m \cancelto{0}{v^2_{t=0}} - m g \cancelto{z_0}{z_{t=0}} \\
    &= \int_{t'=0}^{t'=t}\delta E_{fric} dt' =
    -\int_{s'=s_0}^{s'=s_t}\mu mg ds' =
    -\mu mg (s_t-\cancelto{0}{s_0})
    \end{aligned}

Speaking in terms of altitude, the energy conservation equation can be
rewritten:

.. math::
    z_0 = z_{t} + \frac{v^2_{t}}{2g} + tan\alpha s_t
   :label: altitude-energy-conservation

This result is illustrated in the following figure.

.. _fig-altitude-energy-conservation:

> .. figure:: _static/energyLineTheory.png

Reviewer: Add a quick description to the caption about what the dots and squares are

    :width: 90%

Let us now consider a system of material blocks flowing down a slope
and obeying to Coulomb friction. We can sum the previous equation

> :numref:`fig-altitude-energy-conservation` of each block after weighting

Reviewer: This is the wrong reference, right?
Furthermore it is not displaying correctly on readthedocs, at least not for me..

> Let us now consider a system of material blocks flowing down a slope
> and obeying to Coulomb friction. We can sum the previous equation
> :numref:`fig-altitude-energy-conservation` of each block after weighting
> it by the block mass. This leads to the mass average energy
> conservation equation:

Reviewer: ```suggestion
Considering a system of material blocks flowing down a slope
with to Coulomb friction: we can sumthe previous equation
:numref:`fig-altitude-energy-conservation` of each block after weighting
it by the block mass. This leads to the mass average energy
conservation equation:
```

.. math::
    \bar{z}_0 = \bar{z}_{t} + \frac{\overline{v^2}_{t}}{2g}
    + tan\alpha \bar{s}_t
   :label: altitude-energy-conservation-avg

where the mass average :math:`\bar{a}` value of a quantity :math:`a` is:

.. math::
    \bar{a} = \frac{\sum_k m_k a_k}{\sum_k m_k}

This means that the mass averaged quantities also follow the same energy

> conservation law when express in terms of altitude. The same figure as in

Reviewer: ```suggestion
conservation law when expressed in terms of altitude. The same figure as in
```

:numref:`fig-altitude-energy-conservation` can be drawn for the center of mass
profile path.

> The aim is now to take advantage of this energy conservation line to evaluate
> the DFA simulation. If we compute for the particles in the simulation, the
> mass averaged path profile and compare it to the :math:`\alpha` line,
> we will be able to compute the error compared to the run-out predicted
> by the energy line as well as the error on the velocity altitude. The following
> figures illustrate the problem.

Reviewer: ```suggestion
The aim is to take advantage of this energy conservation line to evaluate
the DFA simulation. Computing the
mass averaged path profile for the particles in the simulation and comparing it to the :math:`\alpha` line allows to compute the error compared to the energy line run-out. This also applies to the error on the velocity altitude. The following
figures illustrate the concept.
```

.. list-table::

    * -

        .. figure:: _static/energyLinePath.png
            :width: 90%

            View of the avalanche simulation and extracted path

      -

        .. figure:: _static/energyLine.png
            :width: 90%

            Simulation path profile with run-out line, :math:`\alpha`

>             line and energy points

Reviewer: Again, add info about dots and squares

From the different mass averaged simulation quantities and the theoretical
:math:`\alpha` line it is possible to extract four error indicators.
The three first ones are related to the run-out. The intersection between
the :math:`\alpha` line and the mass averaged path profile (or its extrapolation
if the profile is too short) defines the theoretical run-out point.
The horizontal and vertical distance between this point and the end of the path
profile defines a run-out error in meters. The angle difference between
the :math:`\alpha` line angle and the DFA simulation run-out line defines
the run-out angle error.
Finally, the Root Mean Square Error (RMSE) between the :math:`\alpha` line
and the DFA simulation energy points defines an error on the velocity altitude

> :math:`\frac{\overline{v^2}}{2g}`.

Reviewer: I think it would be beneficial to have this as an item list

Limitations and remarks
~~~~~~~~~~~~~~~~~~~~~~~~

> We now have to pay attention to stay in the good range of application of
> this test. Indeed, one of the important hypothesis when developing

Reviewer: ```suggestion
It is essential to stay in the applicable range of
this test. Indeed, one of the important hypothesis when developing
```

the energy solution, is that the material is flowing in the steepest
slope direction. If this hypothesis fails, then it is not possible to develop
the analytic energy solution anymore. In the 3D case, the distance vector
:math:`\mathbf{dl}` traveled by the particles reads
:math:`dl = \frac{ds}{cos(\gamma)}`, where :math:`\gamma` is the angle
between the :math:`\mathbf{dl}` vector and the horizontal plane
which can differ from the slope angle :math:`\theta`. If this happens
it is not possible to differentiate anymore the DFA simulation error from

> the hypothesis fail error. The test has then to be used more qualitatively

Reviewer: I do not get the sentence? (even trying french sentence structure does not help in this case :rofl: )

Author: What I want to say here is that in 3D, we have an error coming from the fact that one of the fundamental hypothesis for the energy solution is not satisfied so the runout error will be a sum of the error coming from the unsatisfied hypothesis and the error coming from our numerics (too big dt, to few particles or bad implementation)

> to compare different DFA simulations.

Reviewer: I would remove the last sentence

It is also possible with this test to observe the effect of terms
such as curvature acceleration, artificial viscosity or pressure gradients.
The curvature acceleration modifies the friction term (depending on topography
curvature and particle velocity). This leads to a mismatch between the
energy solution and the DFA simulation.

> The artificial viscosity can lead to viscous dissipation. This should lead
> to shorter run-outs then what the energy solution predicts.

Reviewer: ```suggestion
Artificial viscosity can lead to viscous dissipation leading
to shorter run-outs then what the energy solution predicts.
```

Finally, the effect of the pressure force can be studied, especially the
effect of the computation options.

Procedure
----------
First, the DFA simulation is ran (in our case using the com1DFA module) on the desired avalanche, saving the particles

> (at least the initial and end particles dictionaries).

Reviewer: ```suggestion
(at least the initial and final particles information).
```

            :width: 90%

            Pile after 19.8s

Energy line test
==================
The Energy line test compares the results of the DFA simulation to a geometrical solution that is related
to the total energy of the system. Solely considering coulomb friction this solution is motivated by the first principle
of energy conservation along a simplified topography. Here friction force only depends on the slope angle.
The analytical run-out is the intersection of the path profile with the alpha line defined by the friction angle.
From the alpha line it is also possible to extract information about the flow mass averaged velocity at any time or
position along the path profile.

> 

Author: Not sure I understand. I think the first sentences are enough

   :label: altitude-energy-conservation-avg

where the mass average :math:`\bar{a}` value of a quantity :math:`a` is:

.. math::
    \bar{a} = \frac{\sum_k m_k a_k}{\sum_k m_k}

This means that the mass averaged quantities also follow the same energy
conservation law when expressed in terms of altitude. The same figure as in
:numref:`fig-altitude-energy-conservation` can be drawn for the center of mass
profile path.

The aim is to take advantage of this energy conservation line to evaluate
the DFA simulation. Computing the
mass averaged path profile for the particles in the simulation and comparing
it to the :math:`\alpha` line allows to compute the error compared to the
energy line run-out. This also applies to the error on the velocity altitude.
The following figures illustrate the concept.

.. list-table::

    * -

        .. figure:: _static/energyLinePath.png
            :width: 90%

            View of the avalanche simulation and extracted path

      -

        .. figure:: _static/energyLine.png
            :width: 90%

            Simulation path profile (dark blue curve and dots)
            with the run-out line (dark blue line and velocity altitude

>             squares), :math:`\alpha` line and energy points

Author: It is just to illustrate the equations. I dont know if the colorbar is essential, plus it will be tricky to create

  * The vertical distance between the run-out point and the end
    of the path profile defines the :math:`\epsilon_z` error in meters.

  * The angle difference between the :math:`\alpha` line angle and the DFA
    simulation run-out line defines the :math:`\epsilon_{\alpha}` angle error.

  * The Root Mean Square Error (RMSE) between the :math:`\alpha` line
    and the DFA simulation energy points defines an error on the
    velocity altitude :math:`\frac{\overline{v^2}}{2g}`.

Limitations and remarks
~~~~~~~~~~~~~~~~~~~~~~~~

It is essential to stay where the assumptions of this test hold.
Indeed, one of the important hypotheses when developing
the energy solution, is that the material is flowing in the steepest
slope direction (i.e. where :math:`dl = \frac{ds}{cos(\theta)}` theta holds).
If this hypothesis fails (as illustrared in :numref:`fig-non-steppest-descent`), then it is not possible
to develop the analytic energy solution anymore. In the 3D case,
the distance vector :math:`\mathbf{dl}` traveled by the particles reads
:math:`dl = \frac{ds}{cos(\gamma)}`, where :math:`\gamma` is the angle
between the :math:`\mathbf{dl}` vector and the horizontal plane

> which can differ from the slope angle :math:`\theta`. In this case,
> the energy solution is not anymore the solution of the problem considered
> and can not be used as reference.
> 

Reviewer: ```suggestion
which can differ from the slope angle :math:`\theta`. In this case,
the energy solution is not the solution of the problem anymore
and can not be used as reference.

```
